Homework 02/CM/guesswork_v2.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+***************************************************************************
+    guesswork.py
+
+This is a small guessing game for your computer published at:
+DXDOI:JourLaugh010422/0815/12345
+    ---------------------
+    Date                 : January 2022
+    Copyright            : (C) 2022 by Christian Mielke
+    Email                : christian.mielke@uba.de
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+import random
+
+import numpy
+
+
+class Guestimation:
+    def __init__(self, start: int = 0, stop: int = 100):
+        """
+        Class that contains a number guessing game
+        PC or Human chooses integer between start and top range
+        User or PC needs to do the guessing.
+
+        :inits guestimation object with start,stop and choice attributes
+        :param start: Lowest possible integer for a random integer Defaults to 0
+        :param stop: Highest possible integer for a random integer Defaults to 100
+        :return: options guestimation object
+        """
+        print(
+            f'Welcome to the guessing game either you or your computer needs to guess which number has been chosen out of a randint({start},{stop})')
+        print('\n')
+        self.__choice = None
+        self.start: int = start
+        self.stop: int = stop
+        options: list = ['P', 'H']
+        while True:
```

Comment:
This full `while` loop could be implemented more elegantly:

```suggestion
    while self.gametype not in options:
            self.gametype = input('Please input P for PC guestimation or H for Human guestimation!')
```